product_code/user_service/api/src/main/scala/de/upb/cs/uc4/user/api/UserService.scala

```diff
@@ -82,35 +85,44 @@ trait UserService extends UC4Service {
   /** Allows DELETE, GET, PUT */
   def allowedDeleteGetPut: ServiceCall[NotUsed, Done]
 
+  /** Allows DELETE */
+  def allowedDelete: ServiceCall[NotUsed, Done]
+
   /** Publishes every new user */
   def userCreationTopic(): Topic[EncryptionContainer]
 
-  /** Publishes every deletion of a user */
-  def userDeletionTopic(): Topic[EncryptionContainer]
+  /** Publishes every deletion of a user, sending the username*/
+  def userDeletionTopicMinimal(): Topic[EncryptionContainer]
+
+  /** Publishes every deletion of a user, sending the username and role */
+  def userDeletionTopicPrecise(): Topic[EncryptionContainer]
 
   final override def descriptor: Descriptor = {
     import Service._
     super.descriptor
       .addCalls(
-        restCall(Method.GET, pathPrefix + "/users?usernames", getAllUsers _),
+        restCall(Method.GET, pathPrefix + "/users?usernames&only_active", getAllUsers _),
```

Comment:
Remember to also change the name of the parameter here and in the other restCalls